### PR TITLE
Why use MSFT namespace for DI extensions?

### LINKED ISSUE #20535: Putting my code in a Microsoft namespace
[EDIT by guardrex to add the metadata ..... @ikyriak, it's best to open new issues on the repo using the **This page** feedback button at the bottom of the topic, which adds this metadata automatically and pings the author.]

Related Closed Issue: dotnet/AspNetCore.Docs#11435

Can we have an update regarding the suggestion to "Place extension methods in the Microsoft.Extensions.DependencyInjection namespace to encapsulate groups of service registrations." [[REF](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/dependency-injection.md)].

The last comment was on Jun 27, 2019 from @guardrex : "Engineering has looked this topic over multiple times since then and never said to change it. We'll be reviewing updates again when we get to 3.0 updates. I'll ask explicitly in a few months when I work it."

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0d78f540-81fd-1ba9-cb90-591e4ec27d9b
* Version Independent ID: 2251f557-dd3a-ec35-2558-6243337db009
* Content: [Dependency injection in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-5.0)
* Content Source: [aspnetcore/fundamentals/dependency-injection.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/dependency-injection.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #20535

cc @IEvangelist

### CODE REVIEW

File: aspnetcore/fundamentals/dependency-injection.md [hunk 1]
```diff
@@ -485,7 +485,7 @@ In the sample app, the `IMyDependency` service is registered with the concrete t
 [!code-csharp[](dependency-injection/samples/2.x/DependencyInjectionSample/Startup.cs?name=snippet1&highlight=5)]
 
 > [!NOTE]
-> Each `services.Add{SERVICE_NAME}` extension method adds (and potentially configures) services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations.
+> Each `services.Add{SERVICE_NAME}` extension method adds (and potentially configures) services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods allows them to be displayed in Intellisense.
```
Comment (mkArtakMSFT) [inline aspnetcore/fundamentals/dependency-injection.md]:
```suggestion
> Each `services.Add{SERVICE_NAME}` extension method adds (and potentially configures) services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods allows them to be displayed in Intellisense without any additional `using` blocks added by the developer. Also, it keeps the `Startup` class (where these extension methods are mostly called from) clean from too many `using` blocks.
```

**Review by guardrex — APPROVED** [review]

File: aspnetcore/fundamentals/dependency-injection.md [hunk 2]
```diff
@@ -485,7 +485,7 @@ In the sample app, the `IMyDependency` service is registered with the concrete t
 [!code-csharp[](dependency-injection/samples/2.x/DependencyInjectionSample/Startup.cs?name=snippet1&highlight=5)]
 
 > [!NOTE]
-> Each `services.Add{SERVICE_NAME}` extension method adds (and potentially configures) services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations.
+> Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods also allows them to be displayed in Intellisense.
```
Comment (mkArtakMSFT) [inline aspnetcore/fundamentals/dependency-injection.md:488]:
```suggestion
> Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods also allows them to be displayed in Intellisense without any additional `using` blocks added by the developer.
 Also, it keeps the `Startup` class (where these extension methods are mostly called from) clean from too many `using` blocks.
```
```

**Review by mkArtakMSFT — APPROVED** [review]

File: aspnetcore/fundamentals/dependency-injection.md [hunk 3]
```diff
@@ -485,7 +485,9 @@ In the sample app, the `IMyDependency` service is registered with the concrete t
 [!code-csharp[](dependency-injection/samples/2.x/DependencyInjectionSample/Startup.cs?name=snippet1&highlight=5)]
 
 > [!NOTE]
-> Each `services.Add{SERVICE_NAME}` extension method adds (and potentially configures) services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations.
+> Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddControllersWithViews`, `services.AddRazorPages`, and `services.AddControllers` adds the services ASP.NET Core apps require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Including the namespace portion `Microsoft.Extensions.DependencyInjection` for DI extension methods also:
```
Comment (guardrex) [inline aspnetcore/fundamentals/dependency-injection.md:488]:
```suggestion
> Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddControllersWithViews`, `services.AddRazorPages`, and `services.AddControllers` adds the services ASP.NET Core apps require. We recommended that apps follow this convention. Place extension methods in the <xref:Microsoft.Extensions.DependencyInjection?displayProperty=fullName> namespace to encapsulate groups of service registrations. Including the namespace portion `Microsoft.Extensions.DependencyInjection` for DI extension methods also:
>
```

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @Rick-Anderson !

File: aspnetcore/fundamentals/dependency-injection.md [hunk 4]
```diff
@@ -485,7 +485,9 @@ In the sample app, the `IMyDependency` service is registered with the concrete t
...
+> * Allows them to be displayed in Intellisense with out adding additional `using` blocks.
```
Comment (guardrex) [inline aspnetcore/fundamentals/dependency-injection.md:489]:
```suggestion
> * Allows them to be displayed in [IntelliSense](/visualstudio/ide/using-intellisense) with out adding additional `using` blocks.
```

**Review by guardrex — APPROVED** [review]

**Pull request merged**